File: Codeforces/1324B.cpp
#include <bits/stdc++.h>
#define endl '\n'
using namespace std;

int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    int t,n;
    cin >> t;
    while(t--){
        cin >> n;
        int numbers[n];
        for(int i = 0; i < n; i++){
            cin >> numbers[i];
        }
        bool ans = false;
        for(int i = 0; i < n; i++){
            for(int j = i+2; j < n;j++ ){
                if( numbers[i] == numbers[j]){
                    ans = true;
                    break;
                }
            }
        }
        if(ans){
            cout << "YES" << endl;
        }
        else{
            cout << "NO" << endl;
        }
    }
    return 0;

}

File: Codeforces/1324B.py
t = int(input())
for i in range(t):
    n = int(input())
    quest = [int(i) for i in input().split()]
    ans = False
    end = 0
    for j in range(0,n-2):
        for k in range(j+2,n):
            if quest[j] == quest[k]:
                ans = True
                break
    print("YES" if ans else "NO")
    